pennylane/beta/tapes/tape.py

```diff
@@ -1284,6 +1279,9 @@ def jacobian(self, device, params=None, **options):
         >>> tape.jacobian(dev)
         array([], shape=(4, 0), dtype=float64)
         """
+        if any([m.return_type is State for m in self.measurements]):
+            raise ValueError("The jacobian method does not support circuits that return the state")
```

Comment:
One day it might be cool to consider this :laughing:

Reply:
Yes! We already have done some of the groundwork for the parameter shift rule.

Reply:
Something to keep in mind as we evolve the parameter-shift rule --- allowing gate grad recipes to be altered based on the circuit output.